core/lib/multivm/src/utils/bytecode.rs

```diff
@@ -0,0 +1,154 @@
+use std::collections::HashMap;
+
+use zksync_types::ethabi::{self, Token};
+use zksync_utils::bytecode::{hash_bytecode, validate_bytecode, InvalidBytecodeError};
```

Comment:
Ideally, I'd like to deal with `zksync_utils` so that it doesn't gather functionality across all domains, but this is beyond the scope of this PR.

Reply:
Absolutely.